js/src/lib.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019-2020 Centrality Investments Limited
+
+//! Provide JS-Rust API bindings to create and inspect Cennznut
+use cennznut::{
+    v0::{contract::Contract, method::Method, module::Module, CENNZnutV0},
+    CENNZnut,
+};
+use pact::interpreter::types::PactType;
+use parity_scale_codec::{Decode, Encode};
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_namespace = console)]
+    fn log(s: &str);
+}
+
+#[inline]
+fn from_slice_32(bytes: &[u8]) -> [u8; 32] {
+    let mut array = [0; 32];
+    if bytes.len() < 32 {
+        log("expected 32 byte array");
+        return array;
+    }
+    let bytes = &bytes[..array.len()]; // panics if not enough data
+    array.copy_from_slice(bytes);
+    array
+}
+
+/// A js handle for a rust versioned cennznut struct
+#[wasm_bindgen(js_name = CENNZnut)]
+pub struct JsHandle(CENNZnut);
+
+#[wasm_bindgen(js_class = CENNZnut)]
+#[allow(irrefutable_let_patterns)]
+impl JsHandle {
+    #[wasm_bindgen(constructor)]
+    /// Create a new Cennznut, it is always v0 for now
+    pub fn new(modules: &JsValue, contracts: &JsValue) -> Self {
+        let modules_vec: Vec<(String, Module)> = modules
+            .into_serde()
+            .expect("Deserialization of module failed");
+        let contract_vec: Vec<([u8; 32], Contract)> = contracts
+            .into_serde()
+            .expect("Deserialization of contract failed");
+        let cennznut: CENNZnutV0 = CENNZnutV0 {
+            modules: modules_vec,
+            contracts: contract_vec,
+        };
+        JsHandle(CENNZnut::V0(cennznut))
+    }
+
+    #[allow(non_snake_case)]
+    /// Create modules used for creating Cennznut instance
+    pub fn createModules(moduleName: &str, block_cooldown: u32, methods: &JsValue) -> JsValue {
```

Comment:
`createModule` / should this be a constructor?

Reply:
ya.. we don't need this now.. we can initialise through constructor